src/pyqasm/visitor.py

```diff
@@ -1939,9 +1939,63 @@ def _visit_function_call(
 
         return return_value, result
 
-    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> None:
+    # Custom exceptions for break/continue
+    class BreakException(Exception):
```

Comment:
Can we move these into the [`src/pyqasm/exceptions.py`](https://github.com/qBraid/pyqasm/blob/7076d46e057085fbb6189dbe52d6e6ab4b2b2170/src/pyqasm/exceptions.py) ? 
NOTE : these are not errors, so let us not base them with the `PyqasmError` class. Something like `LoopException` would be apt as the base class